safir/src/safir/metrics/_event_manager.py

```diff
@@ -227,16 +228,13 @@ async def create_publisher(
             raise DuplicateEventError(name)
         payload_model.validate_structure()
 
-        class MetaBase(AvroBaseModel):
-            # Used by dataclasses-avroschema to set the avro name and namespace
-            class Meta:
-                schema_name = name
-                namespace = self._topic
-
         # The full, metadata-enriched event model
         event_class = create_model(
             "EventModel",
-            __base__=(payload_model, EventMetadata, MetaBase),
+            __base__=(payload_model, EventMetadata),
+        )
+        event_class._metadata = SchemaMetadata(  # noqa: SLF001
+            schema_name=name, namespace=self._topic
```

Comment:
Here's another possible way to deal with this, without futzing with internals (though it's the same in spirit, knowing about the internals :( ):
https://github.com/lsst-sqre/safir/pull/309